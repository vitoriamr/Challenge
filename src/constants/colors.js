export const LIGHT_GRAY = "#eaecf2";
export const LINE_GRAY = "#eeeeee";
