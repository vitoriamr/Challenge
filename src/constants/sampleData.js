export const DEFAULT_VALUE = "Select an Option...";

export const STATES = {
  NY: "New York",
  CA: "California",
  TX: "Texas"
};

export const CITIES = {
  NY_CITY: "New York City",
  BUFFALO: "Buffalo",
  LA: "Los Angeles",
  SF: "San Francisco",
  AUSTIN: "Austin",
  SA: "San Antonio"
};
