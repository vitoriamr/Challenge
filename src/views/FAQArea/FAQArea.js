import React, { useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import faqArrowIcon from "../../icons/faqArrow.svg";
import { getFAQContext } from "./FAQconfig/FAQconfig";

const FAQArea = ({ faqVisible, value }) => {
  const [openAnswer, setOpenAnswer] = useState(false);
  const [FAQItem, setFAQItem] = useState(undefined);

  const values = getFAQContext(value);

  const shouldOpenAnswer = (openAnswer, FAQItem, currentFAQ) =>
    openAnswer && FAQItem === currentFAQ;

  return (
    <div>
      <FAQTitle>{faqVisible ? value + " FAQs" : "City FAQs"}</FAQTitle>
      {!faqVisible && (
        <FAQSubtitle>{"Select a State and City to see Info"}</FAQSubtitle>
      )}
      {faqVisible && (
        <QuestionsList>
          {values.map(value => (
            <Grid key={value.item}>
              <Button
                id={`button-${value.item}`}
                onClick={() => {
                  value.item === FAQItem
                    ? setOpenAnswer(!openAnswer)
                    : setOpenAnswer(true);
                  setFAQItem(value.item);
                }}
              >
                <Icon
                  facingUp={shouldOpenAnswer(openAnswer, FAQItem, value.item)}
                  src={faqArrowIcon}
                  alt={"faqArrowIcon"}
                />
              </Button>
              <Question
                shouldOpenAnswer={shouldOpenAnswer(
                  openAnswer,
                  FAQItem,
                  value.item
                )}
              >
                {value.question}
              </Question>
              {shouldOpenAnswer(openAnswer, FAQItem, value.item) && (
                <Answer id={`answer-${value.item}`}>{value.answer} </Answer>
              )}
            </Grid>
          ))}
        </QuestionsList>
      )}
    </div>
  );
};
FAQArea.propTypes = {
  faqVisible: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired
};

const FAQTitle = styled.div`
  font-size: 18px;
  font-family: "Cinzel";
`;
FAQTitle.displayName = "FAQTitle";

const FAQSubtitle = styled.div`
  font-weight: bold;
  margin-top: 10px;
`;
FAQSubtitle.displayName = "FAQSubtitle";

const QuestionsList = styled.div`
  margin-top: 20px;
  width: 500px;
`;
QuestionsList.displayName = "QuestionsList";

const Grid = styled.div`
  display: grid;
  grid-template-columns: 80px auto;
  grid-template-rows: repeat(2, auto);
  grid-template-areas:
    "button question"
    ". answer";
  margin-bottom: 20px;
`;

const Question = styled.div`
  grid-area: question;
  font-weight: bold;
  margin: 8px 0 10px;
  text-align: start;
`;
Question.displayName = "Question";

const Button = styled.button`
  grid-area: button;
  border: none;
  background: transparent;
  height: 30px;
  width: 30px;
  outline: none;
`;
Button.displayName = "Button";

const Icon = styled.img`
  height: 30px;
  width: 30px;
  ${props => (props.facingUp ? "transform: rotate(90deg);" : "")}
`;

const Answer = styled.div`
  grid-area: answer;
  margin-bottom: 20px;
`;
Answer.displayName = "Answer";

export default FAQArea;
