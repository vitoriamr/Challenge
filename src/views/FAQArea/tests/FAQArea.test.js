import React from "react";
import { shallow, mount } from "enzyme";

import { CITIES } from "../../../constants/sampleData";
import FAQArea from "../FAQArea";

describe("FAQArea", () => {
  const requiredProps = {
    faqVisible: true,
    value: CITIES.AUSTIN
  };

  it("matches previous snapshot", () => {
    const wrapper = shallow(<FAQArea {...requiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("FAQ Visible", () => {
    it("should render FAQTitle with value", () => {
      const wrapper = shallow(<FAQArea {...requiredProps} />);
      expect(wrapper.find("FAQTitle").text()).toBe(
        `${requiredProps.value} FAQs`
      );
    });

    it("should not render FAQSubtitle", () => {
      const wrapper = shallow(<FAQArea {...requiredProps} />);
      expect(wrapper.find("FAQSubtitle").exists()).toBe(false);
    });

    it("should render QuestionsList", () => {
      const wrapper = shallow(<FAQArea {...requiredProps} />);
      expect(wrapper.find("QuestionsList").exists()).toBe(true);
    });

    it("should render 2 Questions", () => {
      const wrapper = mount(<FAQArea {...requiredProps} />);
      expect(wrapper.find("Question").length).toBe(2);
    });

    it("should render corresponding answer if button is clicked", () => {
      const wrapper = mount(<FAQArea {...requiredProps} />);
      wrapper.find(`Button[id="button-1"]`).simulate("click");
      expect(wrapper.find(`Answer[id="answer-1"]`).exists()).toBe(true);
    });
  });

  describe("FAQ not Visible", () => {
    const props = {
      ...requiredProps,
      faqVisible: false
    };

    it("should render default FAQTitle", () => {
      const wrapper = shallow(<FAQArea {...props} />);
      expect(wrapper.find("FAQTitle").text()).toBe("City FAQs");
    });

    it("should render FAQSubtitle if faq is not visible", () => {
      const wrapper = shallow(<FAQArea {...props} />);
      expect(wrapper.find("FAQSubtitle").text()).toBe(
        "Select a State and City to see Info"
      );
    });

    it("should not render QuestionsList", () => {
      const wrapper = shallow(<FAQArea {...props} />);
      expect(wrapper.find("QuestionsList").exists()).toBe(false);
    });
  });
});
