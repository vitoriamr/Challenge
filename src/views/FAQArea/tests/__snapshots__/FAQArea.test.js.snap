// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQArea matches previous snapshot 1`] = `
<div>
  <FAQTitle>
    Austin FAQs
  </FAQTitle>
  <QuestionsList>
    <styled.div
      key="1"
    >
      <Button
        id="button-1"
        onClick={[Function]}
      >
        <styled.img
          alt="faqArrowIcon"
          facingUp={false}
          src=""
        />
      </Button>
      <Question
        shouldOpenAnswer={false}
      >
        Where can I find the best breakfast tacos?
      </Question>
    </styled.div>
    <styled.div
      key="2"
    >
      <Button
        id="button-2"
        onClick={[Function]}
      >
        <styled.img
          alt="faqArrowIcon"
          facingUp={false}
          src=""
        />
      </Button>
      <Question
        shouldOpenAnswer={false}
      >
        Is the BBQ really that good?
      </Question>
    </styled.div>
  </QuestionsList>
</div>
`;
